src/wallet/test/spend_tests.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <policy/fees.h>
+#include <validation.h>
+#include <wallet/coincontrol.h>
+#include <wallet/test/util.h>
+#include <wallet/test/wallet_test_fixture.h>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(spend_tests, WalletTestingSetup)
+
+BOOST_FIXTURE_TEST_CASE(SubtractFee, TestChain100Setup)
+{
+    CreateAndProcessBlock({}, GetScriptForRawPubKey(coinbaseKey.GetPubKey()));
+    auto wallet = CreateSyncedWallet(*m_node.chain, m_node.chainman->ActiveChain(), coinbaseKey);
+
+    // Check that a subtract-from-recipient transaction slightly less than the
+    // coinbase input amount does not create a change output (because it would
+    // be uneconomical to add and spend the output), and make sure it pays the
+    // leftover input amount which would have been change to the recipient
+    // instead of the miner.
+    auto check_tx = [&wallet](CAmount leftover_input_amount) {
+        CRecipient recipient{GetScriptForRawPubKey({}), 50 * COIN - leftover_input_amount, true /* subtract fee */};
+        CTransactionRef tx;
+        CAmount fee;
+        int change_pos = -1;
+        bilingual_str error;
+        CCoinControl coin_control;
+        coin_control.m_feerate.emplace(10000);
+        coin_control.fOverrideFeeRate = true;
+        FeeCalculation fee_calc;
+        BOOST_CHECK(wallet->CreateTransaction({recipient}, tx, fee, change_pos, error, coin_control, fee_calc));
+        BOOST_CHECK_EQUAL(tx->vout.size(), 1);
+        BOOST_CHECK_EQUAL(tx->vout[0].nValue, recipient.nAmount + leftover_input_amount - fee);
```

Comment:
The thinking is that the recipient always gets the whole UTXO - fee.
On the first look it seemed to me that the value received would be different since `leftover_input_amount` is different for different invocations. It took me some time to figure out that the `leftover_input_amount` variable  cancels out.

I understand your thinking as well and I don't know which one is better.